client/src/pages/ProfileListings/ProfileListings.tsx

```diff
@@ -0,0 +1,164 @@
+        searchResults.push(user);
+        setDisplayedUsers(searchResults);
+      }
+    });
+  };
+
+  //   Updates based on date selection
+  useEffect(() => {
+    if (formattedDateRange === 'any') {
+      setDisplayedUsers(users.slice(0, 6));
+    } else {
+      if (dateRange.from && dateRange.to) {
+        const searchResults: User[] = [];
+        setDisplayedUsers([]);
+        const range = moment.range(dateRange.from, dateRange.to);
+        users.map((user) => {
+          //   If at least one available date falls within the selected range, the user
+          //   is added to the displayedUsers array.
+          const isInDateRange = (date: Date) => {
+            return range.contains(date);
+          };
+          if (user.availableDates.some(isInDateRange)) {
+            searchResults.push(user);
+            setDisplayedUsers(searchResults);
+          }
+        });
+      }
+    }
+  }, [formattedDateRange, setFormattedDateRange]);
+
+  //   Updates based on search parameters
+  useEffect(() => {
+    if (search) {
+      updateSearch();
+    } else {
+      setDisplayedUsers(users.slice(0, 6));
+    }
+  }, [search, setSearch]);
+
+  const profileCardGrid = (
+    <Grid container className={classes.profilesContainer} xs={9}>
+      {displayedUsers.map((user) => (
+        <ProfileCard user={user} key={user._id} />
+      ))}
+      <Snackbar open={snackbarOpen} autoHideDuration={2000} onClose={handleSnackbarClose}>
+        <Alert severity="info">No more users found</Alert>
+      </Snackbar>
+    </Grid>
+  );
+
+  return (
+    <Grid container className={classes.root}>
+      <Typography variant="h4" className={classes.title}>
+        Your search results
+      </Typography>
+      <Grid>
+        <Grid container className={classes.searchDateContainer}>
+          <Grid className={classes.search}>
+            <SearchIcon className={classes.searchIcon} />
+            <InputBase
+              placeholder="Search by city…"
+              inputProps={{ 'aria-label': 'search' }}
+              onChange={(e) => {
+                setSearch(e.target.value);
+              }}
+            />
+          </Grid>
+          <Grid>
+            <Divider orientation="vertical" />
+          </Grid>
+          <Grid className={classes.dateRangeContainer}>
+            <Button onClick={handleCalendarOpen}>
+              <DateRangeIcon className={classes.dateRangeIcon} />
+            </Button>
+            <Typography>{formattedDateRange.charAt(0).toUpperCase() + formattedDateRange.slice(1)}</Typography>
+            <Button onClick={() => setFormattedDateRange('any')}>
+              <CloseIcon className={classes.dateRangeIcon} />
+            </Button>
+          </Grid>
+        </Grid>
+      </Grid>
+      {profileCardGrid}
+      {search === '' && formattedDateRange === 'any' ? (
+        <Button onClick={handleShowMore} variant="outlined" style={{ margin: '1rem', marginBottom: '2rem' }}>
```

Comment:
try to avoid inline styling, to stay consistent with the architecture of the project.